pyfar/dsp/dsp.py

```diff
@@ -811,6 +813,338 @@ def regularized_spectrum_inversion(
+        simply return the previous or next value of the point; ``'nearest-up'``
+        and ``'nearest'`` differ when interpolating half-integers
+        (e.g. 0.5, 1.5) in that ``'nearest-up'`` rounds up and ``'nearest'``
+        rounds down. The interpolation is done using
+        ``scipy.interpolate.interp1d``.
+    fscale : string, optional
+
+        ``'linear'``
+            Interpolate on a linear frequency axis.
+        ``'log'``
+            Interpolate on a logarithmic frequency axis. Note that 0 Hz can
+            not be interpolated on a logarithmic scale because the logarithm
+            of 0 does not exist. Frequencies of 0 Hz are thus replaced by the
+            next highest frequency before interpolation.
+
+        The default is ``'linear'``.
+    clip : bool, tuple
+        The interpolated magnitude response is clipped to the range specified
+        by this two element tuple. E.g., ``clip=(0, 1)`` will assure that no
+        values smaller than 0 and larger than 1 occur in the interpolated
+        magnitude response. The clipping is applied after the interpolation
+        but before applying linear or minimum phase (in case `method` is
+        ``'magnitude_linear'`` or ``'magnitude_minimum'``. The default is
+        ``False`` which does not clip the
+        data.
+    group_delay : float, optional
+        The group delay of the linear phase response. Must be specified if
+        `method` is ``'magnitude_linear'``. The unit of the group delay is
+        given by `unit` (see below).
+    unit : string, optional
+        Unit of the group delay. Can be ``'s'`` for seconds, ``'ms'`` for
+        milliseconds, ``'mus'`` for microseconds, or ``'samples'``. The
+        default is ``'samples'``.
+
+    Returns
+    -------
+    interpolator : :py:class:`interpolate_spectrum`
+        The interpolator can be called to interpolate the data (see examples
+        below). It returns a :py:class:`~pyfar.classes.audio.Signal` and has
+        the following parameters
+
+        `n_samples` : int
+            Length of the interpolated time signal in samples
+        `sampling_rate`: int
+            Sampling rate of the output signal in Hz
+        `show` : bool, optional
+            Show a plot of the input and output data. The default is ``False``.
+
+    Examples
+    --------
+    Interpolate magnitude only and add artificial linear phase.
+
+    >>> import pyfar as pf
+    >>>
+    >>> data = pf.FrequencyData([1, 0], [5e3, 20e3])
+    >>> interpolator = pf.dsp.interpolate_spectrum(
+    >>>     data, 'magnitude_linear', ('nearest', 'linear', 'nearest'),
+    >>>     group_delay=32)
+    >>> signal = interpolator(64, 44100)
+
+    Inspect the data. Note that this plot can be also created by the
+    interpolator object by ``signal = interpolator(64, 44100, show=True)``
+
+    >>> with pf.plot.context():
+    >>>     _, ax = plt.subplots(2, 2)
+    >>>
+    >>>     # time signal (linear amplitude)
```

Comment:
```suggestion
```
I'd remove these in-line comments and replace with text explaining what's going on above

Reply:
I added information to the text above, removed some comments and kept them where I felt they aid a quicker understanding of the code.